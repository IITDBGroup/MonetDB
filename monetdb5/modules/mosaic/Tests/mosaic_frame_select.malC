b:= bat.new(:timestamp);
bat.append(b,"2014-08-14 12:00:00":timestamp);
bat.append(b,"2014-08-14 12:00:00":timestamp);
bat.append(b,"2014-08-14 12:01:00":timestamp);
bat.append(b,"2014-08-14 12:02:00":timestamp);
bat.append(b,"2014-08-14 12:00:00":timestamp);
bat.append(b,"2014-08-14 12:00:00":timestamp);
bat.append(b,"2014-08-14 12:00:00":timestamp);
bat.append(b,"2014-08-14 12:01:00":timestamp);
bat.append(b,"2014-08-14 12:02:00":timestamp);
bat.append(b,"2014-08-14 12:00:00":timestamp);
b:= bat.append(b,b);
base:= algebra.copy(b);

mosaic.compress(b,"frame");
b:= mosaic.decompress(b);
io.print(base, b);

s:= algebra.select(base,"2014-08-14 12:00:00":timestamp,"2014-08-14 12:02:00":timestamp,false,false,false);
xs:= mosaic.select(b,"2014-08-14 12:00:00":timestamp,"2014-08-14 12:02:00":timestamp,false,false,false);
io.print(s, xs);

s:= algebra.select(base,"2014-08-14 12:00:00":timestamp,"2014-08-14 12:02:00":timestamp,true,false,false);
xs:= mosaic.select(b,"2014-08-14 12:00:00":timestamp,"2014-08-14 12:02:00":timestamp,true,false,false);
io.print(s, xs);

s:= algebra.select(base,"2014-08-14 12:00:00":timestamp,"2014-08-14 12:02:00":timestamp,false,true,false);
xs:= mosaic.select(b,"2014-08-14 12:00:00":timestamp,"2014-08-14 12:02:00":timestamp,false,true,false);
io.print(s, xs);

s:= algebra.select(base,"2014-08-14 12:00:00":timestamp,"2014-08-14 12:02:00":timestamp,true,true,false);
xs:= mosaic.select(b,"2014-08-14 12:00:00":timestamp,"2014-08-14 12:02:00":timestamp,true,true,false);
io.print(s, xs);

s:= algebra.select(base,"2014-08-14 12:00:00":timestamp,"2014-08-14 12:02:00":timestamp,false,false,true);
xs:= mosaic.select(b,"2014-08-14 12:00:00":timestamp,"2014-08-14 12:02:00":timestamp,false,false,true);
io.print(s, xs);

s:= algebra.select(base,"2014-08-14 12:00:00":timestamp,"2014-08-14 12:02:00":timestamp,true,false,true);
xs:= mosaic.select(b,"2014-08-14 12:00:00":timestamp,"2014-08-14 12:02:00":timestamp,true,false,true);
io.print(s, xs);

s:= algebra.select(base,"2014-08-14 12:00:00":timestamp,"2014-08-14 12:02:00":timestamp,false,true,true);
xs:= mosaic.select(b,"2014-08-14 12:00:00":timestamp,"2014-08-14 12:02:00":timestamp,false,true,true);
io.print(s, xs);

s:= algebra.select(base,"2014-08-14 12:00:00":timestamp,"2014-08-14 12:02:00":timestamp,true,true,true);
xs:= mosaic.select(b,"2014-08-14 12:00:00":timestamp,"2014-08-14 12:02:00":timestamp,true,true,true);
io.print(s, xs);
