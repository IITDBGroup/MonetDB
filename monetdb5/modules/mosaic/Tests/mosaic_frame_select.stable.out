stdout of test 'mosaic_frame_select` in directory 'monetdb5/modules/mosaic` itself:


# 07:18:47 >  
# 07:18:47 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=34551" "--set" "mapi_usock=/var/tmp/mtest-1480/.s.monetdb.34551" "--set" "monet_prompt=" "--forcemito" "--set" "mal_listing=2" "--dbpath=/export/scratch1/mk/mosaic//Linux/var/MonetDB/mTests_monetdb5_modules_mal" "mosaic_frame_subselect.mal"
# 07:18:47 >  

# MonetDB 5 server v11.22.0
# This is an unreleased version
# Serving database 'mTests_monetdb5_modules_mal', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs and 128bit integers dynamically linked
# Found 15.590 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2015 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://vienna.ins.cwi.nl:34551/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-1480/.s.monetdb.34551
# MonetDB/GIS module loaded
# Start processing logs sql/sql_logs version 52200
# Start reading the write-ahead log 'sql_logs/sql/log.4'
# Finished reading the write-ahead log 'sql_logs/sql/log.4'
# Finished processing logs sql/sql_logs
# MonetDB/SQL module loaded
# MonetDB/R   module loaded

# 11:50:51 >  
# 11:50:51 >  "mclient" "-lmal" "-ftest" "-tnone" "-Eutf-8" "--host=/var/tmp/mtest-30332" "--port=39586"
# 11:50:51 >  

#--------------------------#
# h	t  # name
# void	void  # type
#--------------------------#
[ 0@0,	2014-08-14 12:00:00.000000,	2014-08-14 12:00:00.000000	]
[ 1@0,	2014-08-14 12:00:00.000000,	2014-08-14 12:00:00.000000	]
[ 2@0,	2014-08-14 12:01:00.000000,	2014-08-14 12:01:00.000000	]
[ 3@0,	2014-08-14 12:02:00.000000,	2014-08-14 12:02:00.000000	]
[ 4@0,	2014-08-14 12:00:00.000000,	2014-08-14 12:00:00.000000	]
[ 5@0,	2014-08-14 12:00:00.000000,	2014-08-14 12:00:00.000000	]
[ 6@0,	2014-08-14 12:00:00.000000,	2014-08-14 12:00:00.000000	]
[ 7@0,	2014-08-14 12:01:00.000000,	2014-08-14 12:01:00.000000	]
[ 8@0,	2014-08-14 12:02:00.000000,	2014-08-14 12:02:00.000000	]
[ 9@0,	2014-08-14 12:00:00.000000,	2014-08-14 12:00:00.000000	]
[ 10@0,	2014-08-14 12:00:00.000000,	2014-08-14 12:00:00.000000	]
[ 11@0,	2014-08-14 12:00:00.000000,	2014-08-14 12:00:00.000000	]
[ 12@0,	2014-08-14 12:01:00.000000,	2014-08-14 12:01:00.000000	]
[ 13@0,	2014-08-14 12:02:00.000000,	2014-08-14 12:02:00.000000	]
[ 14@0,	2014-08-14 12:00:00.000000,	2014-08-14 12:00:00.000000	]
[ 15@0,	2014-08-14 12:00:00.000000,	2014-08-14 12:00:00.000000	]
[ 16@0,	2014-08-14 12:00:00.000000,	2014-08-14 12:00:00.000000	]
[ 17@0,	2014-08-14 12:01:00.000000,	2014-08-14 12:01:00.000000	]
[ 18@0,	2014-08-14 12:02:00.000000,	2014-08-14 12:02:00.000000	]
[ 19@0,	2014-08-14 12:00:00.000000,	2014-08-14 12:00:00.000000	]
#--------------------------#
# t	t	t  # name
# void	oid	oid  # type
#--------------------------#
[ 0@0,	2@0,	2@0	]
[ 1@0,	7@0,	7@0	]
[ 2@0,	12@0,	12@0	]
[ 3@0,	17@0,	17@0	]
#--------------------------#
# t	t	t  # name
# void	oid	oid  # type
#--------------------------#
[ 0@0,	0@0,	0@0	]
[ 1@0,	1@0,	1@0	]
[ 2@0,	2@0,	2@0	]
[ 3@0,	4@0,	4@0	]
[ 4@0,	5@0,	5@0	]
[ 5@0,	6@0,	6@0	]
[ 6@0,	7@0,	7@0	]
[ 7@0,	9@0,	9@0	]
[ 8@0,	10@0,	10@0	]
[ 9@0,	11@0,	11@0	]
[ 10@0,	12@0,	12@0	]
[ 11@0,	14@0,	14@0	]
[ 12@0,	15@0,	15@0	]
[ 13@0,	16@0,	16@0	]
[ 14@0,	17@0,	17@0	]
[ 15@0,	19@0,	19@0	]
#--------------------------#
# t	t	t  # name
# void	oid	oid  # type
#--------------------------#
[ 0@0,	2@0,	2@0	]
[ 1@0,	3@0,	3@0	]
[ 2@0,	7@0,	7@0	]
[ 3@0,	8@0,	8@0	]
[ 4@0,	12@0,	12@0	]
[ 5@0,	13@0,	13@0	]
[ 6@0,	17@0,	17@0	]
[ 7@0,	18@0,	18@0	]
#--------------------------#
# t	t	t  # name
# void	void	oid  # type
#--------------------------#
[ 0@0,	0@0,	0@0	]
[ 1@0,	1@0,	1@0	]
[ 2@0,	2@0,	2@0	]
[ 3@0,	3@0,	3@0	]
[ 4@0,	4@0,	4@0	]
[ 5@0,	5@0,	5@0	]
[ 6@0,	6@0,	6@0	]
[ 7@0,	7@0,	7@0	]
[ 8@0,	8@0,	8@0	]
[ 9@0,	9@0,	9@0	]
[ 10@0,	10@0,	10@0	]
[ 11@0,	11@0,	11@0	]
[ 12@0,	12@0,	12@0	]
[ 13@0,	13@0,	13@0	]
[ 14@0,	14@0,	14@0	]
[ 15@0,	15@0,	15@0	]
[ 16@0,	16@0,	16@0	]
[ 17@0,	17@0,	17@0	]
[ 18@0,	18@0,	18@0	]
[ 19@0,	19@0,	19@0	]
#--------------------------#
# t	t	t  # name
# void	oid	oid  # type
#--------------------------#
[ 0@0,	0@0,	0@0	]
[ 1@0,	1@0,	1@0	]
[ 2@0,	3@0,	3@0	]
[ 3@0,	4@0,	4@0	]
[ 4@0,	5@0,	5@0	]
[ 5@0,	6@0,	6@0	]
[ 6@0,	8@0,	8@0	]
[ 7@0,	9@0,	9@0	]
[ 8@0,	10@0,	10@0	]
[ 9@0,	11@0,	11@0	]
[ 10@0,	13@0,	13@0	]
[ 11@0,	14@0,	14@0	]
[ 12@0,	15@0,	15@0	]
[ 13@0,	16@0,	16@0	]
[ 14@0,	18@0,	18@0	]
[ 15@0,	19@0,	19@0	]
#--------------------------#
# t	t	t  # name
# void	oid	oid  # type
#--------------------------#
[ 0@0,	3@0,	3@0	]
[ 1@0,	8@0,	8@0	]
[ 2@0,	13@0,	13@0	]
[ 3@0,	18@0,	18@0	]
#--------------------------#
# t	t	t  # name
# void	oid	oid  # type
#--------------------------#
[ 0@0,	0@0,	0@0	]
[ 1@0,	1@0,	1@0	]
[ 2@0,	4@0,	4@0	]
[ 3@0,	5@0,	5@0	]
[ 4@0,	6@0,	6@0	]
[ 5@0,	9@0,	9@0	]
[ 6@0,	10@0,	10@0	]
[ 7@0,	11@0,	11@0	]
[ 8@0,	14@0,	14@0	]
[ 9@0,	15@0,	15@0	]
[ 10@0,	16@0,	16@0	]
[ 11@0,	19@0,	19@0	]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#

# 07:18:47 >  
# 07:18:47 >  "Done."
# 07:18:47 >  

